docs/reference/components/aws.cue

```diff
@@ -0,0 +1,153 @@
+package metadata
```

Comment:
Moved from `docs/reference/components/sinks/aws.cue` and updated to cover both sources and sinks.